### PR TITLE
We don't need this redirects anymore. Some of them are wrong

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

We had redirects to cover for early changes in the site structure. We don't need them anymore. The structure has changed even more which makes some of the redirects redundant.

I also update Docusaurus to the next version and cleaned up configs.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
~~- [] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.~~

### CODE REVIEW

File: docs/components/data/features.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ export const features = [
     },
     js: {
       status: ':heavy_check_mark:',
-      docsLink: `[docs](${baseUrl}/js/features/odata/use-odata-v2-type-safe-client-for-javascript-typescript)`,
+      docsLink: `[docs](${baseUrl}/js/features/odata/type-safe-client-odata-v2.mdx)`,
```
Comment (Johannes-Schneider) [inline docs/components/data/features.js:59]: This does not work when I'm testing it locally.

File: docs/components/data/features.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ export const features = [
     },
     js: {
       status: ':heavy_check_mark:',
-      docsLink: `[docs](${baseUrl}/js/features/odata/use-odata-v4-type-safe-client-for-javascript-typescript)`,
+      docsLink: `[docs](${baseUrl}/js/features/odata/type-safe-client-odata-v4.mdx)`,
```
Comment (Johannes-Schneider) [inline docs/components/data/features.js:73]: This does not work when I'm testing it locally.

File: docs/js/features/odata/overview.mdx [hunk 3]
```diff
@@ -18,8 +18,8 @@ keywords:
 
 | OData protocol version                                              | SAP Cloud SDK for JavaScript |
 | ------------------------------------------------------------------- | ---------------------------- |
```
Comment (Johannes-Schneider) [inline docs/js/features/odata/overview.mdx:20]: Could you please adjust the spacing here?

**Review by Johannes-Schneider — CHANGES_REQUESTED** [review]
The two JS links in the feature matrix do not work.
Everything else looks good to me.

File: docs/components/data/features.js (hunk 2 above)
Comment (artemkovalyov) [reply docs/components/data/features.js:73]: WoW, thanks for the diligent review, @Johannes-Schneider

File: docs/js/features/odata/overview.mdx (hunk 3 above)
Comment (artemkovalyov) [reply docs/js/features/odata/overview.mdx:20]: I'll run prettier, it should fix it.

**Review by FrankEssenberger — APPROVED** [review]
LGTM - however, we should have a pipeline step that ensures the local startup is working right? Johannes manual test is not save enough.

**Comment by artemkovalyov** [comment]
> LGTM - however, we should have a pipeline step that ensures the local startup is working right? Johannes manual test is not save enough.

@FrankEssenberger, we run a build on every PR which should catch everything for the static site. The release pipeline does deploy step after merge. If it fails new release is not deployed despite the merge. It means we can ensure the site will run.

If you're about links, they should be checked automatically. I'll check the configs because before broken links were caught automatically.

File: docs/components/data/features.js (hunk 1 above)
Comment (artemkovalyov) [reply docs/components/data/features.js:59]:
```suggestion
      docsLink: `[docs](${baseUrl}/js/features/odata/use-odata-v2-type-safe-client-for-javascript-typescript)`,
```

File: docs/components/data/features.js (hunk 2 above)
Comment (artemkovalyov) [reply docs/components/data/features.js:73]:
```suggestion
      docsLink: `[docs](${baseUrl}/js/features/odata/use-odata-v4-type-safe-client-for-javascript-typescript)`,
```

**Review by Johannes-Schneider — APPROVED** [review]
LGTM.

**Comment by artemkovalyov** [comment]
@Johannes-Schneider, @FrankEssenberger, I found the reason the broken links were not identified. They are generated via the feature matrix and avoid regular check Docusarus runs when processing MDX. For that same reason, a slug (id) has to be used instead of the file name.

**Pull request merged**